plugins/typescript.js

```diff
@@ -247,6 +247,13 @@ module.exports = {
     // requires type annotations to exist
     '@typescript-eslint/typedef': 2,
     // warns for any two overloads that could be unified into one by using a union or an optional/rest parameter
-    '@typescript-eslint/unified-signatures': 2
+    '@typescript-eslint/unified-signatures': 2,
+    // disallows using the non-null assertion on the left operand of the nullish coalescing operator
+    '@typescript-eslint/no-non-null-asserted-nullish-coalescing': 2,
+    // allows to specify new option `allowTypeImport` for path or pattern
+    '@typescript-eslint/no-restricted-imports': [2, {
+      paths: [],
+      patterns: []
+    }]
```

Comment:
Please note that `no-restricted-imports` is an [Extension Rule](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules). Correct handling for an extension is to zero-out the base eslint rule at the top of this file, and then add the corresponding `@typescript-eslint/no-restricted-imports` rule with the very same syntax that we see in https://github.com/zendeskgarden/eslint-config/blob/7fc518180db0af809b017e476bb732a84089d4e3/rules/es6.js#L34-L35

Reply:
and ditto re: alphabetical :wink:

Reply:
Noted, thanks for sharing this.

Reply:
fwiw, it might still be a nice touch to add the `no-restricted-imports: 0` to the top of this file just as a documented indicator that we're going with the `typescript` version of the rule now. Even though they are all `0` throughout 😛